integrations/ambient_scroll.py

```diff
@@ -0,0 +1,45 @@
+import requests
+from constants.chains import Chain
+from constants.integration_ids import IntegrationID
+from models.integration import Integration
+from constants.summary_columns import SummaryColumn
+from constants.ambient import AMBIENT_SCROLL_DEPLOYMENT_BLOCK, AMBIENT_SCROLL_API_URL
+
+
+class Ambient(Integration):
+    def __init__(self):
+        super().__init__(
+            IntegrationID.AMBIENT_SCROLL_USDE_LP,
+            AMBIENT_SCROLL_DEPLOYMENT_BLOCK,
+            Chain.SCROLL,
+            [SummaryColumn.AMBIENT_SCROLL_SHARDS],
+            20,  # TODO: Change 20 to the sats multiplier for the protocol that has been agreed upon
+            1,
+        )
+
+    def get_balance(self, user: str, block: int) -> float:
+        """
+        Get the balance of a user at a given block
+        """
+        url = f"{AMBIENT_SCROLL_API_URL}/sats/scroll/balance"
+        params = {"user": user, "block": block}
+        response = requests.get(url, params=params)
+        data = response.json()
+        return data["balance"]
```

Comment:
I tested this locally and works, but what if the user has both USDe & sUSDe deposited? 
What would the balance return? The sum of these 2 values? 
We need the separate USDe and sUSDe amounts for each user.

Reply:
the balance is normalized to USD, so a pair of 100 USDe + 98 sUSDe will return 198. i can separate it :)

Reply:
Thx for that👍! We need the two separated values because USDe and sUSDe receive different sats multipliers

Reply:
a quick question - shall we still normalize the position to USD, and how should we handle the USDe<>sUSDe pair in this case?

right now, a position like 321 USDC + 123 USDe is counted as ~444 (since Ambient supports concentrated liquidity), so should we count 100 USDe + 98 sUSDe in this case as ~200 for both USDe and sUSDe balances, or separate them?

(asking because i think this might create asymmetry in the rewarded sats, since USDC/USDe pair will be counted as all USDe, while USDe/sUSDe counted as half USDe and half sUSDe, and the values are different)

Reply:
I think that we don't need to normalize the position to USD.
Is it possible that your endpoint returns the token0 amount and token1 amount for the pair separately? And then we can handle everything internally

For example:
- The `USDC<>USDe` pair would return `token0: 321 (USDC), token1: 123 (USDe) `
- The `USDe<>sUSDe` pair would return `token0: 100 (USDe), token1: 98 (sUSDe)`

And would also be very helpful if you could also return which token is the `token0` and `token1` so we know if its USDC, USDe or sUSDe. If easier you could return instead the pool pair e.g `USDC/USDe`

Thanks!

Reply:
ahh i see, so you would like to basically have the function to return a list of positions and their sizes? i assumed that we wanted counting in a single unit because that was the function signatures in the template indicated :)

so if i understand correctly, the `get_balance` method should return a list of dictionaries basically, something like

```
[
  { usdc: 123, usde: 321},
  { usde: 245, susde: 6969}
]
```

and then we separate into 2 classes, one for USDe positions and the other one for sUSDe?

Reply:
Exactly!
We would need to be able to get the position sizes based on blocks, is there a way to retrieve this positions performing onchain calls? e.g `balanceOf` or any other method that we could use to query it?

Reply:
If the 1st option is accurate and we can get the position at any given block via in/outflows that works I think

Reply:
sorry i think i wasn't clear 🙈 - the 1st option if we count the both sides as a whole is fairly accurate and is what we initially did (but we didn't provide separated balance and calculated everything as a sum), but because users can set up an unbalanced range, the ratio between USDe and the other token might change quite a lot (but the total USD value doesn't change much, unless USDe heavily depegs)

if this works, then i will set up the API to return for each position 2 lists, something like:

```
usde_pairs = [{ "usde_usdc": 123, "usde_usdt": 321}]
susde_pairs = [{ "susde_usdc": 123, "susde_usdt": 321}]
```

let me know!

(if this doesnt work, we can also set up the 3rd option but it will take a lot more time)

Reply:
Ah sorry about that, got it now.
Okay I think that works for now, the only issue that I see is with the `USDe<>sUSDe` pair as I would need to know exactly the USDe and the sUSDe amount because USDe and sUSDe receive different sats multiplier, so counting that pair as a whole won't work.
Maybe we can use the 1st option for every pair except for `USDe<>sUSDe` and for this one we use the 3rd?

Reply:
got it, the issue is that if we set up one pair for the 3rd option then it's almost the same complexity as setting it up for all the pairs

may i ask how this API is being used, as in, is it supposed to be real-time or is it allowed to have a few minutes of delay, or the sats reward is actually revealed weekly or only after the program completes? if it's not real time and this is only used to calculate the points per block retroactively, we can also batch it from our side to make it more efficient

Reply:
I see, so it's allowed to have a few hours of delay. 
We run the sats script daily at a random block taking a snapshot and update the user sats. Users are able to see the sats rewards that they are accumulating on our website (this is updated daily as well).

Does that work for you?

Reply:
sounds good, we will do that then. thank you!